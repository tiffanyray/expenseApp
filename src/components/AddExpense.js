import React from 'react';

const AddExpense = () => (
    <div>
        <h1>This is my add expense component</h1>
    </div>
);

export default AddExpense;