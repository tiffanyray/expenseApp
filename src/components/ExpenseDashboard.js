import React from 'react';

const ExpenseDashboardPage = () => (
    <div>
        <h1>This is from dashboard component</h1>
    </div>
);

export default ExpenseDashboardPage;