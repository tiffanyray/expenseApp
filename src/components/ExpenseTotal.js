import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenseTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';


export const ExpenseTotal = ({ expenseCount, expenseTotal }) => {
    const formattedExpensesTotal = numeral(expenseTotal).format('$0,0.00');
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';

    return (
        <div>
            <p>You are viewing {expenseCount} {expenseWord} that total {formattedExpensesTotal}</p>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    

    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: selectExpenseTotal(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpenseTotal);