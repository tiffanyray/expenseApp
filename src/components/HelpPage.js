import React from 'react';

const HelpPage = () => (
    <div>
        <h1>This is my help component</h1>
    </div>
);

export default HelpPage;