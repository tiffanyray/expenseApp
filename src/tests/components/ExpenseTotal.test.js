import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseTotal } from '../../components/ExpenseTotal';

test('should render Expenses Total with one expense', () => {
    const wrapper = shallow(<ExpenseTotal expenseCount={1} expenseTotal={125} />);
    expect(wrapper).toMatchSnapshot();
});


test('should render Expenses Total with multiple expenses', () => {
    const wrapper = shallow(<ExpenseTotal expenseCount={3} expenseTotal={598} />);
    expect(wrapper).toMatchSnapshot();
});

