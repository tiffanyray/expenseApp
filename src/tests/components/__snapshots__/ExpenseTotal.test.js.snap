// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expenses Total with multiple expenses 1`] = `
<div>
  <p>
    You are viewing 
    3
     
    expenses
     that total 
    $598.00
  </p>
</div>
`;

exports[`should render Expenses Total with one expense 1`] = `
<div>
  <p>
    You are viewing 
    1
     
    expense
     that total 
    $125.00
  </p>
</div>
`;
